src/components/Contexts/CollaboratorsContext.tsx

```diff
@@ -0,0 +1,334 @@
+  resetCollaborators: () => void;
+  loadPotentialCollaborators: () => void;
+  loading: boolean;
+  error: ApolloError | null;
+};
+
+/**
+ * CollaboratorsContext to manage collaborators state
+ */
+const CollaboratorsContext = createContext<CollaboratorsCtxState | undefined>(undefined);
+
+/**
+ * Custom hook to use the CollaboratorsContext
+ *
+ * @note Do NOT use this context directly. This is exported for testing purposes only.
+ * @see {@link CollaboratorsProvider}
+ * @see {@link CollaboratorsCtxState}
+ * @throws Error if used outside of CollaboratorsProvider
+ */
+export const useCollaboratorsContext = (): CollaboratorsCtxState => {
+  const context = useContext(CollaboratorsContext);
+  if (!context) {
+    throw new Error("useCollaboratorsContext must be used within a CollaboratorsProvider");
+  }
+  return context;
+};
+
+/**
+ * Default empty collaborator object
+ */
+const defaultEmptyCollaborator: Collaborator = {
+  collaboratorID: "",
+  permission: "Can View",
+} as Collaborator;
+
+type ProviderProps = {
+  children: ReactNode;
+};
+
+/**
+ * CollaboratorsProvider component to provide collaborators context
+ *
+ * @see {@link useSubmissionContext} The context hook
+ * @returns {JSX.Element}
+ */
+export const CollaboratorsProvider: FC<ProviderProps> = ({ children }) => {
+  const { data: submissionData } = useSubmissionContext();
+  const { enqueueSnackbar } = useSnackbar();
+
+  const [potentialCollaborators, setPotentialCollaborators] = useState<Collaborator[]>([]);
+  const [maxCollaborators, setMaxCollaborators] = useState<number>(0);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<ApolloError | null>(null);
+  const [currentCollaborators, setCurrentCollaborators] = useState<Collaborator[]>([
+    { ...defaultEmptyCollaborator },
+  ]);
+
+  const submissionID = submissionData?.getSubmission?._id;
+
+  const [loadPotentialCollaboratorsQuery] = useLazyQuery<
+    ListPotentialCollaboratorsResp,
+    ListPotentialCollaboratorsInput
+  >(LIST_POTENTIAL_COLLABORATORS, {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data) => {
+      setMaxCollaborators(data?.listPotentialCollaborators?.length || 0);
+      const collaborators = data?.listPotentialCollaborators?.map((user) =>
+        userToCollaborator(user)
+      );
+      setPotentialCollaborators(collaborators);
+      setLoading(false);
+    },
+    onError: (err) => {
+      setError(err);
+      setLoading(false);
+    },
+  });
+
+  const [editSubmissionCollaborators, { loading: editLoading }] = useMutation<
+    EditSubmissionCollaboratorsResp,
+    EditSubmissionCollaboratorsInput
+  >(EDIT_SUBMISSION_COLLABORATORS, {
+    fetchPolicy: "no-cache",
+  });
+
+  /**
+   * Maps the submission's collaborators to include additional details from potential collaborators
+   *
+   * @returns {Collaborator[]} The array of collaborators
+   */
+  const mapSubmissionCollaborators = useCallback((): Collaborator[] => {
```

Comment:
No action is necessarily needed here

----

The API stores the collaboratorName as "Last,First" (no space, which is incorrect), but visually it's formatted correctly. After digging into it, it looks like this function is replacing data in the submission object with the data returned by the listPotentialCollaborators API. On the surface, there isn't necessarily an issue with this, but I'm curious the rationale... aka is this solving for some root issue we need to address?

Reply:
It was originally added when nearly all of the properties for collaborators from the getSubmission API were null, so I was using this to grab the rest of the data from the potential collaborators. Most of that is resolved with the exception of the collaboratorName as you noticed. (BE has a PR for this already) Updated in [2cf4ef7](https://github.com/CBIIT/crdc-datahub-ui/pull/508/commits/2cf4ef7b18e58ac6fc94d44d7e350c81c45070da), which actually made me notice other issues with the tests. Fixed those as well.